tools/src/main/java/org/apache/kafka/tools/GetOffsetShell.java

```diff
@@ -0,0 +1,340 @@
+                .withRequiredArg()
+                .describedAs("topic1:1,topic2:0-3,topic3,topic4:5-,topic5:-3")
+                .ofType(String.class);
+        topicOpt = parser.accepts("topic", "The topic to get the offsets for. It also accepts a regular expression. If not present, all authorized topics are queried. Cannot be used if --topic-partitions is present.")
+                .withRequiredArg()
+                .describedAs("topic")
+                .ofType(String.class);
+        partitionsOpt = parser.accepts("partitions", "Comma separated list of partition ids to get the offsets for. If not present, all partitions of the authorized topics are queried. Cannot be used if --topic-partitions is present.")
+                .withRequiredArg()
+                .describedAs("partition ids")
+                .ofType(String.class);
+        timeOpt = parser.accepts("time", "timestamp of the offsets before that. [Note: No offset is returned, if the timestamp greater than recently committed record timestamp is given.]")
+                .withRequiredArg()
+                .describedAs("<timestamp> / -1 or latest / -2 or earliest / -3 or max-timestamp")
+                .ofType(String.class)
+                .defaultsTo("latest");
+        commandConfigOpt = parser.accepts("command-config", "Property file containing configs to be passed to Admin Client.")
+                .withRequiredArg()
+                .describedAs("config file")
+                .ofType(String.class);
+        excludeInternalTopicsOpt = parser.accepts("exclude-internal-topics", "By default, internal topics are included. If specified, internal topics are excluded.");
+
+        if (args.length == 0) {
+            CommandLineUtils.printUsageAndExit(parser, "An interactive shell for getting topic-partition offsets.");
+        }
+
+        options = parser.parse(args);
+
+        if (options.has(bootstrapServerOpt)) {
+            effectiveBrokerListOpt = bootstrapServerOpt;
+        } else {
+            effectiveBrokerListOpt = brokerListOpt;
+        }
+
+        CommandLineUtils.checkRequiredArgs(parser, options, effectiveBrokerListOpt);
+
+        String brokerList = options.valueOf(effectiveBrokerListOpt);
+
+        ToolsUtils.validatePortOrDie(parser, brokerList);
+    }
+
+    public Map<TopicPartition, Long> fetchOffsets() throws IOException, ExecutionException, InterruptedException {
+        String clientId = "GetOffsetShell";
+        String brokerList = options.valueOf(effectiveBrokerListOpt);
+
+        if (options.has(topicPartitionsOpt) && (options.has(topicOpt) || options.has(partitionsOpt))) {
+            throw new IllegalArgumentException("--topic-partitions cannot be used with --topic or --partitions");
+        }
+
+        boolean excludeInternalTopics = options.has(excludeInternalTopicsOpt);
+        OffsetSpec offsetSpec = parseOffsetSpec(options.valueOf(timeOpt));
+
+        TopicPartitionFilter topicPartitionFilter;
```

Comment:
I didn't find this class, did you forgot to add it.

Reply:
That class is in PR mentioned in description. I hope it will be merged once :)